server/api/transaction-line-items.js

```diff
@@ -0,0 +1,26 @@
+const { transactionLineItems } = require('../api-util/lineItems');
+const { getSdk, handleError, serialize } = require('../api-util/sdk');
+
+module.exports = (req, res) => {
+  const { isOwnListing, listingId, bookingData } = req.body;
+
+  const sdk = getSdk(req, res);
+
+  const listingPromise = isOwnListing
+    ? sdk.ownListings.show({ id: listingId })
+    : sdk.listings.show({ id: listingId });
+
+  listingPromise
+    .then(apiResponse => {
+      const listing = apiResponse.data.data;
+      const lineItems = transactionLineItems(listing, bookingData);
```

Comment:
Hmm, is there any harm in always validating the line items? Then it could be inside the `transactionLineItems` helper.

Reply:
I don't there should be any harm but it's just extra step that is not always needed. If I understood correctly Marketplace API should anyway calculate the lineTotal even if the total is already passed in the API request and check that these values match.

Reply:
Ok, now I understand. It doesn't make sense to add the totals if the line items are sent to the Marketplace API that also adds them. It would be just confusing. And the UI needs those for the line items endpoint.

I think a separate function is cleaner than an option, so your changes in the commit you linked seem good IMO. The other places that use line items are sending those to the Marketplace API, and shouldn't add the line totals. Or am I missing something?

Reply:
Now I checked the code again and realized that I got little confused. Somehow I thought that we are calling the transaction line items endpoint in initiate transaction endpoint too but obviously we are just using the transactionLineItems helper function there. So my bad... (but makes me wonder if we should think about the name of the function again).

Reply:
Yeah, I understand the confusion. Maybe you were confused as the API helper name for the endpoint is the same as the line item util function? I wouldn't rename the functions, most customizers will only have to deal with the util function...